cognite/neat/core/extractors/graph_sheet_to_graph.py

```diff
@@ -0,0 +1,132 @@
+    transformation_rule : TransformationRules
+        Transformation rules
+    """
+    intersection = set(graph_capturing_sheet.keys()).intersection(set(transformation_rule.get_defined_classes()))
+
+    if not intersection:
+        logging.error("Graph capturing sheet is not based on transformation rules! Aborting!")
+        raise ValueError("Graph capturing sheet is not based on transformation rules! Aborting!")
+
+    elif len(intersection) == len(graph_capturing_sheet.keys()):
+        logging.info("All classes in the graph capturing sheet are defined in the transformation rules!")
+
+    elif len(intersection) < len(graph_capturing_sheet.keys()):
+        logging.warning(
+            "Graph capturing sheet contains classes that are not defined in the transformation rules! Proceeding..."
+        )
+        warnings.warn(
+            "Graph capturing sheet contains classes that are not defined in the transformation rules! Proceeding...",
+            stacklevel=2,
+        )
+
+    elif len(intersection) < len(transformation_rule.get_defined_classes()):
+        logging.warning(
+            "Transformation rules contain classes that are not present in the graph capturing sheet! Proceeding..."
+        )
+        warnings.warn(
+            "Transformation rules contain classes that are not present in the graph capturing sheet! Proceeding...",
+            stacklevel=2,
+        )
+
+
+def sheet2graph(
+    graph_capturing_sheet: dict[str, pd.DataFrame], transformation_rule: TransformationRules, separator: str = ","
+):
+    """Converts a graph capturing sheet to an RDF graph
+
+    Parameters
+    ----------
+    graph_capturing_sheet : dict[str, pd.DataFrame]
+        Graph capturing sheet
+    transformation_rule : TransformationRules
+        Transformation rules
+    separator : str, optional
+        Multi value separator, by default ","
+    """
+    # Validation that everything is in order before proceeding
+    validate_if_graph_capturing_sheet_empty(graph_capturing_sheet)
+    validate_rules_graph_pair(graph_capturing_sheet, transformation_rule)
+
+    # get class property pairs
+    class_property_pairs = transformation_rule.get_class_property_pairs()
+
+    # Now create empty graph
+    graph = Graph()
+
+    # Add namespaces
+    for prefix, namespace in transformation_rule.prefixes.items():
+        graph.bind(prefix, namespace)
+    graph.bind(transformation_rule.metadata.prefix, transformation_rule.metadata.namespace)
+
+    # Add triples from graph capturing sheet
+    # iterate over sheets
+    for sheet_name, df in graph_capturing_sheet.items():
+        # iterate over sheet rows
+        for _, row in df.iterrows():
+            if row.identifier is None:
+                logging.warning(f"Missing identifier in sheet {sheet_name} at row {row.name}! Skipping...")
+                warnings.warn(
+                    f"Missing identifier in sheet {sheet_name} at row {row.name}! Skipping...",
```

Comment:
Duplicated message.

Reply:
Done

Reply:
Are you missing push? I still see the duplicated messages.